File: homework/src/main/java/com/mikheev/homework/service/BookService.java
package com.mikheev.homework.service;

import com.mikheev.homework.domain.Book;

import java.util.List;

public interface BookService {

    List<Book> getAllBooks();

    Book getBookWithAllAssociations(long id);

    void addBook(Book book);

    Book updateBook(Book book);

    void deleteBook(long id);
}

File: homework/src/main/kotlin/com/mikheev/homework/service/BookService.kt
package com.mikheev.homework.service

import com.mikheev.homework.domain.Book

interface BookService {

    fun getAllBooks(): List<Book>
    fun getBookWithAllAssociations(id: Long): Book
    fun addBook(book: Book): Book
    fun updateBook(book: Book): Book
    fun deleteBook(id: Long)
}
